File: startup/src/main/java/code/startup/services/ContaService.java
package code.startup.services;

import java.util.List;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import code.startup.models.Conta;
import code.startup.models.Funcionario;
import code.startup.repositories.ContaRepository;

@Service
public class ContaService {

	@Autowired
	private ContaRepository contaRepository;	

	@Autowired
	private FuncionarioService funcionarioService;
	
	public List<Conta> mostrarTodasContas(){
		List<Conta> conta = contaRepository.findAll();
		return conta;
	}
	
	public Conta mostrarUmaConta(Integer id_conta) {
		Optional<Conta> conta = contaRepository.findById(id_conta);
		return conta.orElseThrow();
	}
	
	public Conta cadastrarConta(Integer id_funcionario, Conta conta) {
		conta.setId_conta(null);
		Funcionario funcionario = funcionarioService.buscarUmFuncionario(id_funcionario);
		conta.setFuncionario(funcionario);
		return contaRepository.save(conta);
	}
	
	public Conta editarConta(Conta conta) {
		mostrarUmaConta(conta.getId_conta());
		return contaRepository.save(conta);
	}
	
	public void deletarConta(Integer id_conta) {
		contaRepository.deleteById(id_conta);
	}
	
	@SuppressWarnings("rawtypes")
	public List<List> buscarContaDosFuncionarios(){
		List<List> contaDosFuncionarios = contaRepository.fetchByContaDosFuncionarios();
		return contaDosFuncionarios;
	}
	
	
}

File: src/app/services/conta.service.ts
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Conta } from '../models/conta';
import { Funcionario } from '../models/funcionario';

@Injectable({
  providedIn: 'root'
})
export class ContaService {

  baseUrl: String = 'http://localhost:8080/startup'

  constructor(private http: HttpClient) { }

  mostrarUmaConta(id_conta: string):Observable<Conta>{
    const url = `${this.baseUrl}/conta/${id_conta}`
    return this.http.get<Conta>(url)
  }
  cadastrarConta(conta: Conta, id_funcionario: string):Observable<Conta>{
    const url = `${this.baseUrl}/conta?funcionario=${id_funcionario}`
    return this.http.post<Conta>(url, conta)
  }

  editarConta(conta: Conta, id_conta: string, id_funcionario: string):Observable<Conta>{
    const url = `${this.baseUrl}/conta/${id_conta}?funcionario=${id_funcionario}`
    return this.http.put<Conta>(url, conta)
  }

  deletarConta(id_conta: string):Observable<void>{
    const url = `${this.baseUrl}/conta/${id_conta}`
    return this.http.delete<void>(url)
  }

  contaDosFuncionarios():Observable<void>{
    const url = `${this.baseUrl}/conta/contas-dos-funcionarios`
    return this.http.get<void>(url)
  }


}
